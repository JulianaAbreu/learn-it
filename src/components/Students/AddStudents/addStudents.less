@import url("../../../assets/styles/_variables.less");

.content-student {
  margin-top: 8%;
  align-content: center;
  align-items: center;
  h1 {
    color: @dark_gray;
  }
  .add-students {
    width: 100%;
    margin-top: 0.5%;
  }
  .btn-student {
    padding: 5px;
    border-radius: 5px;
    border-color: @primary-color;
    font-size: 16px;
    cursor: pointer;
    outline: 0;
  }
  .add-students {
    background-color: #f9f9f9;
    padding: 2%;
    border-radius: 5px;

    .btn-add {
      width: 15%;
      margin-left: 8%;
    }
  }

  .clicked-btn {
    padding: 5px;
    border-radius: 5px;
    border-color: white;
    background-color: @primary-color;
    color: white;
    font-size: 16px;
    cursor: pointer;
    outline: 0;
  }
}
.ant-col-24,
.ant-col-offset-3 {
  width: 100%;
}
.ant-table-wrapper {
  margin-top: 5%;
}
.ant-row,
.content-mainpage {
  width: 100%;
}
