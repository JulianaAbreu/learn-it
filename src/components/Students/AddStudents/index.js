import React, { Component } from "react";
import { Input, InputNumber, DatePicker, Button } from "antd";
import moment from "moment";
import { Row, Col } from "antd";

import "./addStudents.less";

class AddStudents extends Component {
  constructor(props) {
    super(props);
    this.state = {
      on: false
    };
  }

  onClickAddStudent = () => {
    this.setState({
      on: !this.state.on
    });
    console.log(this.state.on);
  };

  render() {
    const onButton = this.state;
    const dateFormat = "DD/MM/YYYY";

    return (
      <div className="content-student">
        <h1>Estudantes</h1>
        <Button type="primary" ghost  className={onButton.on ? 'clicked-btn' : 'btn-student'} onClick={this.onClickAddStudent} >Novo Estudante</Button>

      
        {onButton.on ? (
          <section className="add-students">
            <Row gutter={12}>
              <Col span={6}>
                <Input
                  placeholder="Nome do estudante"
                  className="input-name"
                  Row={2}
                />
              </Col>
              <Col span={3}>
                <Input
                  min={1}
                  max={10}
                  placeholder="31894712"
                  className="input-date"
                />
              </Col>
              <Col span={4}>
                <Input placeholder="3A" className="registration" />
              </Col>
              <DatePicker
                defaultValue={moment("2015/01/01", dateFormat)}
                format={dateFormat}
              />
              <Button type="primary" className="btn-add">
                Inserir
              </Button>
            </Row>
          </section>
        ) : (
          ""
        )}
      </div>
    );
  }
}

export default AddStudents;
