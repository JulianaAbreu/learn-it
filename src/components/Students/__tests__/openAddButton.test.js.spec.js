import React from "react";
import { shallow } from 'enzyme';
import AddStudents from "../AddStudents"
import renderer from 'react-test-renderer';

describe('<AddStudents />', () => {
    it('should not have the filters visible', () => {
        const component = renderer.create(
            <AddStudents />,
        );
        let on = component.toJSON();
        expect(on).toMatchSnapshot();
    })

    it('should Change state of add button', () => {
        const component = renderer.create(
            <AddStudents />,
        );
        let on = component.toJSON();
        expect(on).toMatchSnapshot();
    });

    it('should have a New Student button', () => {
        const wrapper = shallow(<AddStudents />);
        expect(wrapper.find('button.btn-student')).toHaveLength(1);
    })

    it('should click on Novo Estudante button', () => {
        const wrapper = shallow(<AddStudents />);
        wrapper.find('button.btn-student').simulate('click');
        expect(wrapper.state().on).toBe(true);
    })
})
