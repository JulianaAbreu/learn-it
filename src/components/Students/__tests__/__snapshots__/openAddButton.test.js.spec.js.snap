// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddStudents /> should Change state of add button 1`] = `
<div
  className="content-student"
>
  <h1>
    Estudantes
  </h1>
  <button
    className="btn-student"
    onClick={[Function]}
  >
    Novo Estudante
  </button>
  
</div>
`;

exports[`<AddStudents /> should not have the filters visible 1`] = `
<div
  className="content-student"
>
  <h1>
    Estudantes
  </h1>
  <button
    className="btn-student"
    onClick={[Function]}
  >
    Novo Estudante
  </button>
  
</div>
`;
